jib-cli/src/main/java/com/google/cloud/tools/jib/cli/ContainerBuilders.java

```diff
@@ -0,0 +1,83 @@
+import static com.google.cloud.tools.jib.api.Jib.TAR_IMAGE_PREFIX;
+
+import com.google.cloud.tools.jib.api.DockerDaemonImage;
+import com.google.cloud.tools.jib.api.ImageReference;
+import com.google.cloud.tools.jib.api.InvalidImageReferenceException;
+import com.google.cloud.tools.jib.api.Jib;
+import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.api.RegistryImage;
+import com.google.cloud.tools.jib.api.TarImage;
+import com.google.cloud.tools.jib.api.buildplan.Platform;
+import com.google.cloud.tools.jib.frontend.CredentialRetrieverFactory;
+import com.google.cloud.tools.jib.plugins.common.DefaultCredentialRetrievers;
+import com.google.cloud.tools.jib.plugins.common.logging.ConsoleLogger;
+import java.io.FileNotFoundException;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Helper class for creating JibContainerBuilders from JibCli specifications. */
+public class ContainerBuilders {
+
+  /**
+   * Creates a {@link JibContainerBuilder} depending on the base image specified.
+   *
+   * @param baseImageReference base image
+   * @param platforms platforms for multi-platform support in build command.
+   * @param commonCliOptions common cli options
+   * @param logger console logger
+   * @return a {@link JibContainerBuilder}
+   * @throws InvalidImageReferenceException if the baseImage reference cannot be parsed
+   * @throws FileNotFoundException if credential helper file cannot be found
+   */
+  public static JibContainerBuilder create(
+      String baseImageReference,
+      List<Platform> platforms,
+      CommonCliOptions commonCliOptions,
+      ConsoleLogger logger)
+      throws InvalidImageReferenceException, FileNotFoundException {
+    if (baseImageReference.startsWith(DOCKER_DAEMON_IMAGE_PREFIX)) {
+      return Jib.from(
+          DockerDaemonImage.named(baseImageReference.replaceFirst(DOCKER_DAEMON_IMAGE_PREFIX, "")));
+    }
+    if (baseImageReference.startsWith(TAR_IMAGE_PREFIX)) {
+      return Jib.from(
+          TarImage.at(Paths.get(baseImageReference.replaceFirst(TAR_IMAGE_PREFIX, ""))));
+    }
+    ImageReference imageReference =
+        ImageReference.parse(baseImageReference.replaceFirst(REGISTRY_IMAGE_PREFIX, ""));
+    RegistryImage registryImage = RegistryImage.named(imageReference);
+    DefaultCredentialRetrievers defaultCredentialRetrievers =
+        DefaultCredentialRetrievers.init(
+            CredentialRetrieverFactory.forImage(
+                imageReference,
+                logEvent -> logger.log(logEvent.getLevel(), logEvent.getMessage())));
+    Credentials.getFromCredentialRetrievers(commonCliOptions, defaultCredentialRetrievers)
+        .forEach(registryImage::addCredentialRetriever);
+    JibContainerBuilder containerBuilder = Jib.from(registryImage);
+    if (!platforms.isEmpty()) {
+      containerBuilder.setPlatforms(platforms.stream().collect(Collectors.toSet()));
```

Comment:
If you make `create()` accept `Set<Platform>`, it won't need `stream().collect()`.

BTW, [this link](https://stackoverflow.com/q/27611896/1701388) says `toSet()` doesn't preserve the order. The order of platforms is important, so that's why the original code used `Collectors.toCollection(LinkedHashSet::new))`.

Reply:
Ah my bad, thank you!